File: PetHealth/customer-app/src/main/java/pe/edu/upc/pethealth/network/PetHealthServices.java
package pe.edu.upc.pethealth.network;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;


import pe.edu.upc.lib.ApptModel;
import pe.edu.upc.lib.VeterinaryModel;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PetHealthServices {
    @POST(EndpointUrls.LOGIN)
    Call<RestView<JsonObject>> login(@Body JsonObject requestBody);

    @POST(EndpointUrls.APPOINTMENTS)
    Call<ApptModel.Response> getAppts(@Header("access_token") String accessToken, @Path("user_id") int userId);

    @GET(EndpointUrls.VETERINARIES)
    Call<VeterinaryModel.Response> getCloseVeterinaries(@Header("access_token") String accessToken, @Query("latitude") Double latitude, @Query("longitude") Double longitude);
}

File: PetHealth/customer-app/src/main/java/pe/edu/upc/pethealth/network/PetHealthServices.kt
package pe.edu.upc.pethealth.network

import com.google.gson.JsonObject
import org.json.JSONObject


import pe.edu.upc.lib.models.ApptModel
import pe.edu.upc.lib.models.VeterinaryModel
import retrofit2.Call
import retrofit2.http.*

interface PetHealthServices {
    @POST(EndpointUrls.LOGIN)
    fun login(@Body requestBody: JsonObject): Call<RestView<JsonObject>>

    @GET(EndpointUrls.APPOINTMENTS)
    fun getAppts(@Header("access_token") accessToken: String, @Path("user_id") userId: Int): Call<ApptModel.Response>

    @GET(EndpointUrls.VETERINARIES)
    fun getCloseVeterinaries(@Header("access_token") accessToken: String, @Query("latitude") latitude: Double?, @Query("longitude") longitude: Double?): Call<VeterinaryModel.Response>

    @PUT(EndpointUrls.EDIT_USER)
    fun editProfileData(@Header("access_token") accessToken: String, @Path("user_id") id: Int, @Body requestBody: JSONObject): Call<RestView<JsonObject>>
}
